### PR TITLE
Add trainable into summary, modified the format slightly

### PR DESCRIPTION
**Pull request by noklam**
As request, added Trainable information into model summary.

Usage:
https://gist.github.com/noklam/bdb4e6e3a73f89c7cbbbef7d42140a0b
Previous PR for vision summary: #1237
Dev Project Index: https://forums.fast.ai/t/dev-projects-index/29849
Discussion thread on Forum: https://forums.fast.ai/t/model-summary-implementation/31193

Plan for this PR
add:

- [x] Trainable information in model summary

add:

graph visualization for model structure
Backbone, head structure shown in summary

### CODE REVIEW

**Comment by sgugger** [comment]
Great! Can you just remove commented code that was probably there for debugging?

File: fastai/callbacks/hooks.py [hunk 1]
```diff
@@ -117,10 +117,14 @@ def num_features_model(m:nn.Module)->int:
     return model_sizes(m)[-1][1]
 
 def total_params(m:nn.Module)->int:
-    params = 0
-    if hasattr(m, "weight") and hasattr(m.weight, "size"): params += m.weight.numel()
+    params, trainable = 0, False
+    if hasattr(m, "weight") and hasattr(m.weight, "size"):
+         params += m.weight.numel()
+         trainable = m.weight.requires_grad
+        #  print(m.weight.requires_grad)
```
Comment (sgugger) [inline fastai/callbacks/hooks.py]: Remove

File: fastai/callbacks/hooks.py [hunk 2]
```diff
@@ -136,36 +140,43 @@ def params_size(m: nn.Module, size: tuple = (64, 64))->Tuple[Sizes, Tensor, Hook
...
     hooks = zip(hooks_outputs, hooks_params)
-    res = [(o[0].stored.shape, o[1].stored) for o in hooks]
-    output_size, params = map(list, zip(*res))
-    return (output_size, params, hooks)
+    x = m.eval()(*x) if is_listy(x) else m.eval()(x)
+    output_size = [(o.stored.shape) for o in hooks_outputs]
+    params = [o.stored for o in hooks_params]
+    params, trainables = map(list,zip(*params))
+
+    # import pdb; pdb.set_trace()
```
Comment (sgugger) [inline fastai/callbacks/hooks.py]: Remove with space above

File: fastai/callbacks/hooks.py [hunk 3]
```diff
@@ -117,10 +117,14 @@ def num_features_model(m:nn.Module)->int:
...
-    params = 0
-    if hasattr(m, "weight") and hasattr(m.weight, "size"): params += m.weight.numel()
+    params, trainable = 0, False
+    if hasattr(m, "weight") and hasattr(m.weight, "size"):
+         params += m.weight.numel()
+         trainable = m.weight.requires_grad
+        #  print(m.weight.requires_grad)
     if hasattr(m, "bias") and hasattr(m.bias, "size"):     params += m.bias.numel()
-    return params
+
```
Comment (sgugger) [inline fastai/callbacks/hooks.py]: Remove empty line

**Comment by noklam** [comment]
Cleaned up the debug code and spaces, thanks sgugger .

**Pull request merged**